### ISSUE TITLE
FIXME: Accomodate issue when populating active mod list, stemming from UUIDs in `sprint-a1.0.3`

### ISSUE THREAD

**Issue by twstagg** [oceancabbage/RimSort#43]
* [x] Calculate & track duplicate mods during active mods list creation
* [x] Local mods are loaded first out of any potential duplicates (alphabetically the first mod found by filename) for the "base" packageId that is found in ModsConfig.xml

1. Create 2 instances of an active mod (both with the same packageId) & startup RimSort - it does not matter if these mods are 2 local, 2 steam, 2 expansion, etc
2. Observe that both instances of said mod are populated into the active mods list, even though there should only be one entry in your `ModsConfig.xml` (unless you click save with it in this state; then it would save a <li> with the same packageId for every instance in your duplicate)

This is caused by: `util.mods.populate_active_mods_workshop_data()`

Current version of this function as of this posting:
https://github.com/oceancabbage/RimSort/blob/sprint-a1.0.3/util/mods.py#L1152

```
def populate_active_mods_workshop_data(
    unpopulated_mods: Dict[str, Any], workshop_and_expansions: Dict[str, Any]
) -> Tuple[Dict[str, Any], List[str]]:
    """
    Given a dict of mods with no attached data, populate all
    workshop mods with workshop mods data (taken from the workshop directory).
    Note that key-values in unpopulated mods that cannot find a
    corresponding package id in workshop mods will not have their
    values modified (will be kept empty).
    :param unpopulated_mods: dict of package-id-keyed active mods
    :param workshop_and_expansions: dict of workshop mods (and expansions) keyed to packge-id
    :return: active mod list with populated data, and list of package ids of invalid mods
    """
    logger.info("Populating empty active mods with data")
    populated_mods = unpopulated_mods.copy()
    invalid_mods = []
    for mod_uuid in unpopulated_mods:
        # Cross reference package ids with mods in the workshop folder.
        # If unable to, that means either the mod hasn't been downloaded
        # of it is an invalid entry.
        if (
            unpopulated_mods[mod_uuid]["packageId"]
            == workshop_and_expansions[mod_uuid]["packageId"]
        ):
            populated_mods[mod_uuid] = workshop_and_expansions[mod_uuid]
        else:
            logger.warning(
                f"Unable to find local/workshop data for listed active mod: {mod_uuid}"
            )
            invalid_mods.append(mod_uuid)
            del populated_mods[mod_uuid]
            # populated_mods[mod_package_id] = {"name": f"ERROR({mod_package_id})", "packageId": mod_package_id}
    logger.info("Finished populating empty active mods with data")
    return populated_mods, invalid_mods
```

**Comment by twstagg** [comment]
Fixed as of a9813b5

**Comment by twstagg** [comment]
Fixed in 8159a5d
